### ISSUE TITLE
Detekt Build: configuration cache breaks `--scan`

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5769]
## Expected Behavior
Build scan published based on the existence of `--scan` flag.

## Observed Behavior
(some irrelevant logs are removed so the repro is clean)

Ensure that `./.gradle/configuration-cache/` is empty.

<details><summary>Output of second repro, because that's the most annoying.</summary>

### `./gradlew --scan`
Works as expected, tries to publish build scan to Gradle (not private GE).
```
Configuration cache is an incubating feature.
Calculating task graph as no configuration cache is available for tasks:
Type-safe project accessors is an incubating feature.

> Task :help

Welcome to Gradle 7.6.

To run a build, run gradlew <task> ...

To see a list of available tasks, run gradlew tasks

To see more detail about a task, run gradlew help --task <task>

To see a list of command-line options, run gradlew --help

For more detail on using Gradle, see https://docs.gradle.org/7.6/userguide/command_line_interface.html

For troubleshooting, visit https://help.gradle.org

0 problems were found storing the configuration cache.

See the complete report at file:///P:/projects/contrib/github-detekt/build/reports/configuration-cache/2arwbhktyyefedvjsmav3t9rh/jfig33bfackb5f00ymq15dhg/configuration-cache-report.html

BUILD SUCCESSFUL in 5s
17 actionable tasks: 1 executed, 16 up-to-date

Publishing a build scan to scans.gradle.com requires accepting the Gradle Terms of Service defined at https://gradle.com/terms-of-service. Do you accept these terms? [yes, no] no

Gradle Terms of Service not accepted.

You must answer 'yes' to publish a build scan when prompted on the command line or accept the Gradle Terms of Service in a buildScan configuration block.

For more information, please see https://gradle.com/help/plugin-terms-of-service.

Alternatively, if you are using Gradle Enterprise, specify the server location.
For more information, please see https://gradle.com/help/plugin-enterprise-config.

Configuration cache entry stored.
```

### `./gradlew --scan`
Works as expected, tries to publish build scan to Gradle (not private GE), but config from cache.
```
P:\projects\contrib\github-detekt>
Configuration cache is an incubating feature.
Reusing configuration cache.

> Task :help

Welcome to Gradle 7.6.

To run a build, run gradlew <task> ...

To see a list of available tasks, run gradlew tasks

To see more detail about a task, run gradlew help --task <task>

To see a list of command-line options, run gradlew --help

For more detail on using Gradle, see https://docs.gradle.org/7.6/userguide/command_line_interface.html

For troubleshooting, visit https://help.gradle.org

BUILD SUCCESSFUL in 2s
1 actionable task: 1 executed

Publishing a build scan to scans.gradle.com requires accepting the Gradle Terms of Service defined at https://gradle.com/terms-of-service. Do you accept these terms? [yes, no] no

Gradle Terms of Service not accepted.

You must answer 'yes' to publish a build scan when prompted on the command line or accept the Gradle Terms of Service in a buildScan configuration block.

For more information, please see https://gradle.com/help/plugin-terms-of-service.

Alternatively, if you are using Gradle Enterprise, specify the server location.
For more information, please see https://gradle.com/help/plugin-enterprise-config.

Configuration cache entry reused.
```

### `gradlew`
BAD behavior, tries to publish build scan to Gradle even thought `--scan` is not present.

```
Configuration cache is an incubating feature.
Reusing configuration cache.

> Task :help

Welcome to Gradle 7.6.

To run a build, run gradlew <task> ...

To see a list of available tasks, run gradlew tasks

To see more detail about a task, run gradlew help --task <task>

To see a list of command-line options, run gradlew --help

For more detail on using Gradle, see https://docs.gradle.org/7.6/userguide/command_line_interface.html

For troubleshooting, visit https://help.gradle.org

BUILD SUCCESSFUL in 2s
1 actionable task: 1 executed

Publishing a build scan to scans.gradle.com requires accepting the Gradle Terms of Service defined at https://gradle.com/terms-of-service. Do you accept these terms? [yes, no] no

Gradle Terms of Service not accepted.

You must answer 'yes' to publish a build scan when prompted on the command line or accept the Gradle Terms of Service in a buildScan configuration block.

For more information, please see https://gradle.com/help/plugin-terms-of-service.

Alternatively, if you are using Gradle Enterprise, specify the server location.
For more information, please see https://gradle.com/help/plugin-enterprise-config.

Configuration cache entry reused.
```

</details>

### Simple, but destroys configuration cache (build scan not published)
1. `rm -rf ./gradle/configuration-cache`
2. `gradlew` (no tasks, or any tasks; it's just easier without tasks)  
   Observe: `Calculating task graph as no configuration cache is available for tasks:`
   Observe: `Configuration cache entry stored.`
   Observe: "build scan ..."
    * if you're a contributor: `A build scan was not published as you have not authenticated with server 'ge.detekt.dev'.`
    * if you're a collaborator: a build scan published to GE.
3. `gradlew --scan` (same tasks as in previous step)
   Observe: same as above, exactly the same, even though the `--scan` parameter was given.

@VitalyVPinchuk observed this behavior. (He didn't clear the config cache, that's only needed to make it consistent, it will happen to any contributor who executes a new set of tasks with `--scan`.)

### Simple, but destroys configuration cache (build scan forcefully published)
1. `rm -rf ./gradle/configuration-cache`
2. `gradlew --scan` (no tasks, or any tasks; it's just easier without tasks)  
   Observe: `Calculating task graph as no configuration cache is available for tasks:`
   Observe: `Configuration cache entry stored.`
   Observe: "build scan ..."
    * if you're a contributor:
      ```
      Publishing a build scan to scans.gradle.com requires accepting the Gradle Terms of Service defined at https://gradle.com/terms-of-service. Do you accept these terms? [yes, no] yes
      
      Gradle Terms of Service accepted.
      
      Publishing build scan...
      ```
    * if you're a collaborator: a build scan published to GE.
3. `gradlew` (same tasks as in previous step)
   Observe: same as above, exactly the same, even though the `--scan` parameter was NOT given.

### More complex, but most of the cache is kept.
Same repro as above, except:
 * don't delete the configuration cache
 * choose a set of Gradle tasks that you've never executed, for example `gradlew :random-module:compileKotlin`.

## Context
@VitalyVPinchuk is messing about with Detekt codebase and finds these gems. 😎 (normal, expected usage)

## Your Environment

* Version of detekt used: N/A 0341753f8a7468a7fff5ff37e545f6bceee1f6ba on `main`
* Version of Gradle used (if applicable): 7.6 0341753f8a7468a7fff5ff37e545f6bceee1f6ba on `main`
* Gradle scan link (add `--scan` option when running the gradle task): :trollface:
* Operating System and version: Windows / Mac Ventura 13.2 (22D49)
* Link to your project (if it's a public repository): 0341753f8a7468a7fff5ff37e545f6bceee1f6ba

**Comment by 3flex** [comment]
I only skimmed this because there's a lot of detail, but why is this a bug for detekt? Isn't it an issue with Gradle and/or the enterprise plugin?

**Comment by TWiStErRob** [comment]
In my understanding it is caused by our conditional usage of the startParameters in settings.gradle.kts.

**Comment by 3flex** [comment]
Looks like it may be a Gradle issue after all, someone else has reported what looks like the same thing with a very basic project: https://github.com/gradle/gradle/issues/24023

If you agree that this needs to be fixed upstream we should close this as there's nothing we can do to address it.

**Comment by TWiStErRob** [comment]
Only seems to partially cover; I'm almost certain our conditional commands just add extra on top of the original issue, lets see how they triage it and what they say.

**Comment by atulgpt** [comment]
Hi, @TWiStErRob I checked this locally. It is not reproducing. Seems like this got fixed by gradle update

**Comment by TWiStErRob** [comment]
Indeed, gradle/gradle#24023 has been fixed in 8.1.
